src/clock_estimator_uust1.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2024 The urg_stamped Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <vector>
+
+#include <ros/time.h>
+
+#include <urg_stamped/device_state_estimator.h>
+#include <scip2/logger.h>
+
+namespace urg_stamped
+{
+namespace device_state_estimator
+{
+
+void ClockEstimatorUUST1::startSync()
+{
+  sync_samples_.clear();
+  cnt_dropped_samples_ = 0;
+}
+
+void ClockEstimatorUUST1::pushSyncSample(
+    const ros::Time& t_req, const ros::Time& t_res, const uint64_t device_wall_stamp)
+{
+  const SyncSampleUUST1 s(t_req, t_res, device_wall_stamp);
+  if (s.delay_ > ros::Duration(DEVICE_TIMESTAMP_RESOLUTION))
+  {
+    cnt_dropped_samples_++;
+    return;
+  }
+  sync_samples_.push_back(s);
+  if (comm_delay_.min_.isZero() || comm_delay_.min_ > s.delay_)
+  {
+    comm_delay_.min_ = s.delay_;
+  }
+}
+
+bool ClockEstimatorUUST1::hasEnoughSyncSamples() const
+{
+  const size_t n = sync_samples_.size();
+  if (cnt_dropped_samples_ >= MAX_DROPPED_SAMPLES)
+  {
+    return true;
+  }
+  if (n < MIN_SYNC_SAMPLES)
+  {
+    return false;
+  }
+  if (n >= MAX_SYNC_SAMPLES)
+  {
+    return true;
+  }
+  const OriginFracPart overflow_range = originFracOverflow();
+  return overflow_range.t_max_ > overflow_range.t_min_;
```

Comment:
Should we check if `overflow_range` is valid i.e. `return overflow_range && overflow_range.t_max_ > overflow_range.t_min_;`?

Reply:
As this seems be a bug in the master branch, I'll fix it in a new PR

Reply:
`overflow_range` is an instance and `t_max_ = t_min_ = 0` when invalid. So, the result is correct even if `originFracOverflow()` failed